### PR TITLE
KokkosKernels: remove static constexpr member (#8418)

### LINKED ISSUE #8418: Tpetra: Bad interaction with Kokkos ETI with XL compiler
## Bug Report
@trilinos/tpetra
@jhux2 
@brian-kelley 

### Description
SPARC is failing to link on vortex pwr9 seemingly because of the change in TpetraExt_MatrixMatrix_def.hpp from 75333e4ed. The XL compiler is the only place it has appeared. The error is as follows: 

```
/projects/atdm_devops/trilinos_installs/2020-12-02/ats2-xl-2020.03.18_spmpi-rolling_serial_static_opt/lib/libtpetraext.a(TpetraExt_MatrixMatrix.cpp.o):(.toc+0x68): undefined reference to `KokkosSparse::Experimental::SortedCountEntries<unsigned long, int, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<unsigned long*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> > >::ORDINAL_MAX'
```

Please confirm that the deduced template arguments are the intended template arguments when using `GO=long long` and `LO = int`.

### PR DESCRIPTION
**Pull request by brian-kelley**
Remove static constexpr member variables from SpADD functors.
In official C++ pre 17, these need out-of-class definitions just like
non-constexpr static members, but I didn't add them here.
GCC/Clang let this work anyway, but XL goes by the
standard so this fixes building with XL.

Here's what I don't understand: Kokkos_ArithTraits.hpp and KokkosSparse_gauss_seidel_impl.hpp both also have static constexpr member variables, but those didn't cause linker errors with XL. They also didn't show up as symbols in executables, whereas this "ORDINAL_MAX" does. Not sure what's going on, but I didn't do a workaround for those since they aren't causing problems.

@trilinos/kokkos-kernels 

## Related Issues

* Closes #8418 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 

## Testing

I replicated #8418 on vortex, and this fixes it. Tpetra's MatrixMatrix tests bulid and run successfully.

### CODE REVIEW

File: packages/kokkos-kernels/src/sparse/KokkosSparse_spadd.hpp [hunk 1]
```diff
@@ -202,67 +202,6 @@ struct UnmergedSumFunctor {
...
-        size_type, typename CcolindsT::non_const_value_type,
-        typename CcolindsT::non_const_value_type, TeamMember>(
-        Ccolinds.data() + rowStart, ABperm.data() + rowStart, rowNum, t);
-  }
-  CrowptrsT Crowptrs;
-  CcolindsT Ccolinds;
-  CcolindsT ABperm;
-};
-#endif
-
```
Comment (jhux2) [inline packages/kokkos-kernels/src/sparse/KokkosSparse_spadd.hpp:265]: @brian-kelley Why was it necessary to remove this?

**Review by jhux2 — APPROVED** [review]

**Review by ndellingwood — APPROVED** [review]
lgtm, thanks @brian-kelley !
Noting here that `SortEntriesFunctor` removal is simply cleanup and unrelated to the changes needed to address the static constexpr link issues with xl (already cleaned up in kokkos-kernels' develop branch)

**Comment by ndellingwood** [comment]
Cross-referencing kokkos/kokkos-kernels#859 (that PR will clobber these changes with the 3.3 release)

File: packages/kokkos-kernels/src/sparse/KokkosSparse_spadd.hpp (hunk 1 above)
Comment (brian-kelley) [reply packages/kokkos-kernels/src/sparse/KokkosSparse_spadd.hpp:265]: Oops, I missed that when I copied from KokkosKernels - but like Nathan said it's not used anymore (replaced by ``sort_crs_matrix()``).

**Comment by jhux2** [comment]
@brian-kelley Thank you for getting a patch in so quickly.

**Pull request merged**